nexus/lib/qmcpack.py

```diff
@@ -509,6 +511,224 @@ def post_analyze(self,analyzer):
+                        for idx,(tw,bnd) in enumerate(zip(dsc.TwistIndex,dsc.BandIndex)):
+                            if tw == int(tw1) and bnd == int(bnd1):
+                                # This orbital should no longer be in the set of occupied orbitals
+                                if idx<elns.groups[spin_channel[0]].size:
+                                    msg  = 'WARNING: You requested \'{}\' excitation of type \'{}\',\n'
+                                    msg += '         however, the first orbital \'{} {}\' is still occupied (see einspline file).\n'
+                                    msg += '         Please check your input.'
+                                    msg = msg.format(spin_channel,exc_input[1],tw1,bnd1)
+                                    exc_failure = True
+                                #end if
+                            elif tw == int(tw2) and bnd == int(bnd2):
+                                # This orbital should be in the set of occupied orbitals
+                                if idx>=elns.groups[spin_channel[0]].size:
+                                    msg  = 'WARNING: You requested \'{}\' excitation of type \'{}\',\n'
+                                    msg += '         however, the second orbital \'{} {}\' is not occupied (see einspline file).\n'
+                                    msg += '         Please check your input.'
+                                    msg = msg.format(spin_channel,exc_input[1],tw2,bnd2)
+                                    exc_failure = True
+                                #end if
+                            #end if
+                        #end for
+                    else:
+                        self.warn('No check for \'{}\' excitation of type \'{}\' was done. When this path is possible, then a check should be written.'.format(exc_input[0],exc_input[1]))
+                    #end if
+                elif exc_type in (exc_types.energy,exc_types.lowest):
+                    # Lowest or Energy Index '-orbindex1 +orbindex2'. Eg., '-4 +5'
+                    if exc_type==exc_types.lowest:
+                        if exc_spin==exc_spins.down:
+                            orb1 = elns.groups.d.size
+                        else:
+                            orb1 = elns.groups.u.size
+                        #end if
+                        orb2 = orb1+1 
+                    else:
+                        orb1 = int(exc_input[1].split()[0][1:])
+                        orb2 = int(exc_input[1].split()[1][1:])
+                    #end if
+                    if exc1 in ('up','down'):
+
+                        spin_channel = exc1
+                        nelec = elns.groups[spin_channel[0]].size
+
+                        orb1_eig = sorted(edata[spin_channel].Energy)[orb1-1]
+                        orb2_eig = sorted(edata[spin_channel].Energy)[orb2-1]
+
+                        if orb1_eig not in edata[spin_channel]['Energy'][nelec:]:
```

Comment:
Possible precision problem on floating point comparison.
Likely something like the below is better:

etol = 1e-6
if np.abs(eigs_spin[nelec:]-orb1_eig).max() > tol:

Reply:
There was an issue with our solution here if orb1 and orb2 were degenerate. But I think the latest commit resolves this